lib/matplotlib/_types.py

```diff
@@ -0,0 +1,165 @@
+    ``linewidth/2``, away from any point of the center line. However, this
+    results in corners appearing "rounded", which may not be the desired
+    behavior if you are drawing, for example, a polygon or pointed star.
+
+    Matplotlib provides three options for defining how the corners where two
+    segments meet should be drawn. In short:
+
+        - *miter* is the "arrow-tip" style. Each boundary of the filled-in area
+          will extend in a straight line parallel to the tangent vector of the
+          centerline at the point it meets the corner, until they meet in a
+          sharp point.
+        - *round* stokes every point within a radius of ``linewidth/2`` of the
+          center lines.
+        - *bevel* is the "squared-off" style. It can be thought of as a rounded
+          corner where the "circular" part of the corner has been cut off.
+
+    .. note::
+
+        The *miter* option can be controller further by specifying a "miter
+        limit", which specifies how long a miter tip can get before it is
+        automatically "bevel"ed off. Matplotlib does not currently expose a
+        ``miterlimit`` parameter to the user, and most backends simply use the
+        upstream default value. For example, the PDF backend assumes the
+        default value of 10 specified by the PDF standard, while the SVG
+        backend does not even specify the miter limit, resulting in a default
+        value of 4 per the SVG specification.
+
+        A more detailed description of the effect of a miter limit can be found
+        in the `Mozilla Developer Docs
+        <https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-miterlimit>`_
+    """
+
+    miter = 'miter'
+    round = 'round'
+    bevel = 'bevel'
+
+    def __init__(self, s):
+        s = _deprecate_case_insensitive_join_cap(s)
+        Enum.__init__(self)
+
+    @staticmethod
+    def demo():
+        import numpy as np
+        import matplotlib.pyplot as plt
+
+        def plot_angle(ax, x, y, angle, style):
+            phi = np.radians(angle)
+            xx = [x + .5, x, x + .5*np.cos(phi)]
+            yy = [y, y, y + .5*np.sin(phi)]
+            ax.plot(xx, yy, lw=12, color='tab:blue', solid_joinstyle=style)
+            ax.plot(xx, yy, lw=1, color='black')
+            ax.plot(xx[1], yy[1], 'o', color='tab:red', markersize=3)
+
+        fig, ax = plt.subplots(figsize=(8, 6))
+        ax.set_title('Join style')
+        for x, style in enumerate(['miter', 'round', 'bevel']):
+            ax.text(x, 5, style)
+            for y, angle in enumerate([20, 45, 60, 90, 120]):
+                plot_angle(ax, x, y, angle, style)
+                if x == 0:
+                    ax.text(-1.3, y, f'{angle} degrees')
+        ax.set_xlim(-1.5, 2.75)
+        ax.set_ylim(-.5, 5.5)
+        ax.set_axis_off()
+
+        plt.show()
+
+
+class CapStyle(Enum):
+    """
+    Define how the the end of a line is drawn.
```

Comment:
```suggestion
    Define how open end points of a line are drawn.
```

Reply:
While I immediately see what you mean, I don't think "open" vs "closed" will be the most enlightening for a user actually using this docstring to figure out what a `CapStyle` **is**.

Is what I have now better?